mono/tests/sgen-domain-unload-2.cs

```diff
@@ -37,7 +40,32 @@ static void BackgroundNoise ()
 			AppDomain.Unload (ad);
 			Console.Write (".");
 			if (i > 0 && i % 20 == 0) Console.WriteLine ();
+			if (!testHasTimeLeft(startTime)) {
+				var finishTime = DateTime.Now;
+				Console.WriteLine("stopping after {0}", finishTime-startTime);
+				break;
+			}
 		}
 		Console.WriteLine ("\ndone");
 	}
-}
\ No newline at end of file
+
+	// 5s slack
+	private static readonly TimeSpan slack = TimeSpan.FromMilliseconds(5000);
```

Comment:
No idea. unload-2 does manage to print a fair number of dots before it dies, so maybe? :grin: